lib/libzfs/libzfs_mount.c

```diff
@@ -206,6 +226,42 @@ dir_is_empty(const char *dirname)
 }
 
 /*
+ * Returns true if the specified directory is empty.  If we can't open the
+ * directory at all, return true so that the mount can fail with a more
+ * informative error message.
+ */
+static boolean_t
+dir_is_empty(const char *dirname)
+{
+	struct statfs64 st;
+
+	/*
+	 * If the statvfs call fails or the filesystem is not a ZFS
+	 * filesystem, fall back to the slow path which uses readdir.
+	 */
+	if ((statfs64(dirname, &st) != 0) ||
+	    (st.f_type != 0x2fc12fc1)) {
```

Comment:
Right, so this is a little awkward because we want this value to be shared between user and kernel space and ideally would like to avoid  defining it twice.  I'd suggest moving the ZFS_SUPER_MAGIC definition in to `include/sys/fs/zfs.h` right after `#define ZFS_DEV ...`  This header will be picked up by both user and kernel builds.